src/ZkFold/Base/Algebra/Polynomials/Univariate.hs

```diff
@@ -54,7 +55,7 @@ qr a b = go a b zero
         go x y q = if deg x < deg y then (q, x) else go x' y q'
             where
                 c = lt x / lt y
-                n = deg x - deg y
+                n = fromIntegral (deg x - deg y)
```

Comment:
What happens if the difference is negative and hence is not a Natural?

Reply:
If the difference is negative (`deg x < deg y`), `n` is not evaluated at all, see above